src/utils.ts

```diff
@@ -160,6 +185,27 @@ export const serializeValue = <T extends string>(
     }
   }
 
+  if (
+    (part instanceof RequiredEnumArray || part instanceof OptionalEnumArray) &&
+    (typeof value === 'string' || Array.isArray(value))
+  ) {
+    return ([] as readonly (string | number)[]).concat(value).map((sub) => {
+      const validValue = part.valid.find((v) => v.toString() === sub);
+
+      if (validValue !== undefined) {
+        return validValue;
+      } else if (ignoreInvalidEnums) {
+        return sub;
+      }
+
+      throw new Error(`Invalid value for part "${part.name}" - ${value}`);
+    });
```

Comment:
Could you make this a reduce that only includes values that are valid. At the moment you return `sub` when `ignoreInvalidEnums` is true, but this will return invalid values.